addons/mass_mailing/static/src/builder/plugins/social_media_option_plugin.js

```diff
@@ -0,0 +1,324 @@
+        const baseMedias =
+            await this.dependencies.massMailingSocialMediaOptionPlugin.fetchRecordedSocialMedia(
+                company.id
+            );
+        const allLinks = [];
+        for (const [platform, href] of Object.entries(baseMedias)) {
+            const newLinkElement =
+                this.dependencies.massMailingSocialMediaOptionPlugin.renderSocialMediaLink(
+                    platform,
+                    href
+                );
+            allLinks.push(newLinkElement);
+        }
+        if (!allLinks.length) {
+            const placeholder = this.document.createElement("span");
+            placeholder.classList.add("s_social_media_placeholder");
+            placeholder.append(
+                this.document.createTextNode(
+                    _t("Currently no links for %s's social media links", company.name)
+                )
+            );
+            allLinks.push(placeholder);
+        }
+        editingElement.querySelectorAll("a[data-platform]").forEach((element) => {
+            element.remove();
+        });
+        editingElement.firstElementChild.after(...allLinks);
+        editingElement.dataset.companyId = company.id;
+    }
+    isApplied({ editingElement, params: company }) {
+        return parseInt(editingElement.dataset.companyId) === company.id;
+    }
+}
+
+class ToggleSocialMediaLinkAction extends BuilderAction {
+    static id = "toggleSocialMediaLink";
+    static dependencies = ["massMailingSocialMediaOptionPlugin"];
+    apply({ editingElement, params: { platform, getIndex } }) {
+        const editingLink = editingElement.querySelector(`[data-platform="${platform}"]`);
+        const href = this.dependencies.massMailingSocialMediaOptionPlugin.getMedias()[platform];
+        const newLinkElement =
+            this.dependencies.massMailingSocialMediaOptionPlugin.renderSocialMediaLink(
+                platform,
+                href
+            );
+
+        if (editingLink) {
+            editingLink.remove();
+        } else if (getIndex() === undefined) {
+            const firstLink = editingElement.querySelector("[data-platform]:first-of-type");
+            (firstLink ? firstLink.before : editingElement.append)(newLinkElement);
+        } else {
+            const indexedLink = editingElement.querySelector(
+                `[data-platform]:nth-of-type(${getIndex() + 1})`
+            );
+            indexedLink.after(newLinkElement);
+        }
+    }
+    isApplied({ editingElement, params }) {
+        return editingElement.querySelector(`[data-platform="${params.platform}"]`) !== null;
+    }
+}
+
+class RemoveCustomSocialMediaLinkAction extends BuilderAction {
+    static id = "removeCustomMediaLink";
+    apply({ editingElement, params: { platform } }) {
+        const link = editingElement.querySelector(`[data-platform="${platform}"]`);
+        link.remove();
+    }
+}
+
+class AddSocialMediaLinkAction extends BuilderAction {
```

Comment:
There is currently a bug where the placeholder is displayed as well as the links when I insert the social media links